src/main/java/org/jenkins/plugins/lockableresources/queue/Utils.java

```diff
@@ -45,4 +47,30 @@ public static LockableResourcesStruct requiredResources(@NonNull Job<?, ?> proje
 
         return null;
     }
+
+    // check if matrix plugin is installed
+    public static boolean isMatrixPluginEnabled() {
+        PluginWrapper matrixPlugin = Jenkins.get().getPluginManager().getPlugin("matrix-project");
+        // check installation
+        if (matrixPlugin == null) {
+            return false;
+        }
+
+        return matrixPlugin.isEnabled();
+    }
+
+    public static boolean isMatrixProject(Job<?, ?> project) {
+        if (!isMatrixPluginEnabled() || project == null) return false;
+        return project.getClass().getName().equals("hudson.matrix.MatrixProject");
```

Comment:
If you are comparing by `.class.name` then there is no need to also check whether the plugin is active: clearly if the object is a `MatrixProject`, the plugin must be active.